modules/machineset-aws-existing-placement-group.adoc

```diff
@@ -0,0 +1,76 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating-machinesets/creating-machineset-aws.adoc
+// * machine_management/control_plane_machine_management/cpmso-using.adoc
+
+ifeval::["{context}" == "cpmso-using"]
+:cpmso:
+endif::[]
+
+:_content-type: PROCEDURE
+[id="machineset-aws-existing-placement-group_{context}"]
+= Assigning machines to placement groups for Elastic Fabric Adapter instances by using machine sets
+
+You can configure a machine set to deploy machines on link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html[Elastic Fabric Adapter] (EFA) instances within an existing AWS placement group.
+
+EFA instances do not require placement groups, and you can use placement groups for purposes other than configuring an EFA. This example uses both to demonstrate a configuration that can improve network performance for machines within the specified placement group.
+
+.Prerequisites
+
+* You have created a placement group in the AWS console.
++
+[NOTE]
+====
+Ensure that the link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/placement-groups.html#limitations-placement-groups[rules and limitations] for the type of placement group that you create are compatible with your intended use case.
+ifdef::cpmso[]
+The control plane machine set spreads the control plane machines across multiple failure domains when possible. To use placement groups for the control plane, you must use a placement group type that can span multiple Availability Zones.
+endif::cpmso[]
+====
+
+.Procedure
+
+. In a text editor, open the YAML file for an existing machine set or create a new one.
+
+. Edit the following lines under the `providerSpec` field:
++
+[source,yaml]
+----
+ifndef::cpmso[]
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+endif::cpmso[]
+ifdef::cpmso[]
+apiVersion: machine.openshift.io/v1
+kind: ControlPlaneMachineSet
+endif::cpmso[]
+# ...
+spec:
+  template:
+    spec:
+      providerSpec:
+        value:
+          instanceType: <supported_instance_type> # <1>
+          networkInterfaceType: EFA # <2>
+          placement:
+            availabilityZone: <zone> # <3>
+            region: <region> # <4>
+          placementGroupName: <placement_group> # <5>
+# ...
+----
+<1> Specify an instance type that link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html#efa-instance-types[supports EFAs].
+<2> Specify the `EFA` network interface type.
+<3> Specify the zone, for example, `us-east-1a`.
+<4> Specify the region, for example, `us-east-1`.
+<5> Specify the name of the existing AWS placement group to deploy machines in.
+
```

Comment:
I find myself wanting some indication that I should save or wait or whatever's next here--particularly since the procedure right before it in cpmso-using ends with an explicit save step.

Reply:
tl;dr, this is like that for consistency but I agree.

This is something that is in my backlog. Basically I did not add it after thinking about that because I would need to add it to all of them. I then thought it would be better to just ditch this format entirely and make it more like the `install-config.yaml` parameters reference material. I want to do that probably between 4.14 and 4.15.